### ISSUE TITLE
[BUG][Kotlin] Wrong default value is generated if the field type is BigDecimal

### ISSUE THREAD

**Issue by alitams** [OpenAPITools/openapi-generator#10866]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

If one of the type's mapping value is BigDecimal, and sets a default value, then a simple Int is given to the value.

Generated:
```kotlin
val bar: java.math.BigDecimal? = 0
```

Expected: 
```kotlin
val bar: java.math.BigDecimal? = java.math.BigDecimal.ZERO
```

(or BigDecimal(0))

##### openapi-generator version

5.3.0

##### OpenAPI declaration file content or url

```yaml
...
schemas:
  Foo:
      description: 'Xy'
      type: object
      properties:
        bar:
          type: number
          default: 0
```

##### Generation Details
```
java -jar -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005 modules/openapi-generator-cli/target/openapi-generator-cli.jar generate \
   -i xy.yaml \
   -g kotlin-client \
   -o gen/generated \
   --additional-properties=library=jvm-retrofit2,useRxJava3=true,dateLibrary=threetenbp-localdatetime,enumPropertyNaming=UPPERCASE,serializationLibrary=jackson,enablePostProcessFile=true 
```

##### Steps to reproduce

Generate the code below, and check the result.

##### Suggest a fix

The BigDecimal should handled to generate the schema well.

For example:
```kotlin
    override fun toDefaultValue(schema: Schema<*>?): String? {
        val referencedSchema = ModelUtils.getReferencedSchema(this.openAPI, schema)
        val defaultValue =
            if (referencedSchema.default is BigDecimal) {
                getBigDecimalDefaultValue(referencedSchema.default as BigDecimal)
            } else {
                super.toDefaultValue(schema)
            }

        return defaultValue ?: referencedSchema.default?.toString()
    }
```

**Comment by jaycarlton** [comment]
I see the same problem in Java. I was using 
```
    ServiceUsage:
      description: Service Usage Consumer Quota Overrides
      properties:
        bigQueryDailyUsageQuotaOverrideValueBytes:
          description: |-
            Daily maximum query usage per project. Use 0 for no override (default limit is
            effectively unlimited).
          type: number
          format: long
          default: 0
      type: object
 ```
 
 And the generated code tried to do 
 ```
   @JsonProperty("bigQueryDailyUsageQuotaOverrideValueBytes")
  private BigDecimal bigQueryDailyUsageQuotaOverrideValueBytes = 0;
  ```
  which fails at runtime with 
  ```
  error: incompatible types: int cannot be converted to BigDecimal
  private BigDecimal bigQueryDailyUsageQuotaOverrideValueBytes = 0;
  ```

**Comment by sjoblomj** [comment]
@alitams: I have provided a PR to fix this bug in #12213.

@jaycarlton: I am unable to reproduce your results in the Java client. Could you please provide the generator settings that you use when the bug appears?
